File: Api/IntranetWebApi/IntranetWebApi.Domain/Models/Dto/GetPresenceByIdUserDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace IntranetWebApi.Domain.Models.Dto;

public class GetPresenceByIdUserDto
{
    public string Date { get; set; } = null!;
    public int DayNumber { get; set; }
    public bool IsFreeDay { get; set; } = true;
    public bool IsPresent { get; set; }
    public int PresentType { get; set; }
    public string AbsenceReason { get; set; } = null;
    public string StartTime { get; set; } = null!;
    public string EndTime { get; set; } = null!;
    public string WorkHours { get; set; } = null!;
    public string ExtraWorkHours { get; set; } = null!;
}

File: Client/src/app/models/dto/getPresenceByIdUserDto.ts
export interface GetPresenceByIdUserDto {
    date?: string,
    dayNumber?: number,
    isFreeDay?: boolean,
    isPresent?: boolean,
    presentType?: number,
    absenceReason?: string,
    startTime?: string,
    endTime?: string,
    workHours?: string,
    extraWorkHours?: string
}
